api/metadata/models.py

```diff
@@ -0,0 +1,49 @@
+from core.models import AbstractBaseExportableModel
+from django.contrib.contenttypes.fields import GenericForeignKey
+from django.contrib.contenttypes.models import ContentType
+from django.db import models
+
+from organisations.models import Organisation
+
+from .fields import GenericObjectID
+
+FIELD_VALUE_MAX_LENGTH = 2000
+
+
+class FieldType(models.TextChoices):
```

Comment:
We probably want to add more data types here, and make them less closely coupled to python data types. So for example, we probably want to split STRING into SINGLE_LINE_STRING and MULTI_LINE_STRING (primarily for the UI's benefit). 

I haven't looked through the rest of the PR but I'm imagining we would define custom validation functions against the field types.

Reply:
Also we should have a `URL` data type